### PR TITLE
Create help-wanted.md

### PR DESCRIPTION
**Pull request by kvncrw**
initial commit of help-wanted framework

- introduce a guide intended to encourage contributors by specifically labeling issues we feel are good for first-time contributors `contributor/good-first-issue` AND
- introduce a label that indicates the issue is groomed enough for regular contributors and external maintainers to work on independently `contributor/help-wanted`

internal discussions should also take place on how we want to handle the management and grooming of these issues, particularly the cadence, priority, and ownership of these tasks.

### CODE REVIEW

**Comment by notsureifkevin** [comment]
This needs to be formatted using https://sembr.org/

File: help-wanted.md
Comment (jbdoumenjou) [inline help-wanted.md]: As a general comment, I would apply the Keep It Simple Stupid principle.

Instead of trying to define a specific process with a list of constraints/LiveCycle, I would prefer to welcome contributors with a message that gives confidence and some specifics references to start (cf next comments).

One advantage of a message when we apply the label is that the contributor has all the needed information to start to work and doesn't have to be afraid of an external reference or specifics rules to apply. 
The other advantage is to focus on the contributor experience and adapt our behavior to have a personalized experience.

File: help-wanted.md
Comment (jbdoumenjou) [inline help-wanted.md]: For me, this label is for a wanted item but we don't have time to work on it.
The only constraint on it is that we considered that an external contributor could work on it whatever the complexity/context of the task.

We can decide to ask for help whatever the difficulty or context and we can answer with a message that provides the needed information to start to work on it and give confidence.

File: help-wanted.md
Comment (jbdoumenjou) [inline help-wanted.md]: All of this part concerns communication management and does not seem relevant here.

File: help-wanted.md
Comment (jbdoumenjou) [inline help-wanted.md]: In the same way, I would just add a message when we apply the label with something like "We will have special attention and help you with this PR/issue, you can find some doc here and here, don't hesitate to ask for help".
And avoid writing another external document to read.

I think we don't have to apply any process/LiveCycle that takes the number of `good-first-issue` PR into account to know if someone can work on a `contributor/help-wanted`. For me `help-wanted` and `good-first-issue` are not correlated and we can't know the skills nor the energy that the contributor could/would provide.

File: help-wanted.md
Comment (rtribotte) [reply help-wanted.md]: I agree with @jbdoumenjou, as long as we are choosing those issues, the complexity should not be strict criteria, but an element of discussion during the labeling an issue

File: help-wanted.md
Comment (kvncrw) [reply help-wanted.md]: I'm still recovering from illness, so I'm still a bit hazy -- I'll probably hold off on conversating on the other points because I'm unclear on if there are any action items for them.

However, for this one, I believe the goal is to provide communication management advice for both internal AND  external maintainers and contributors. We could put this somewhere in notion, or elsewhere, but it seems relevant here if our goal is to eventually empower members of the community to step up and engage in these activities. WDYT?

File: help-wanted.md
Comment (kvncrw) [reply help-wanted.md]: > We can decide to ask for help whatever the difficulty or context and we can answer with a message that provides the needed information to start to work on it and give confidence.

The idea behind meeting the criteria listed below beforehand is to provide contributors a path to start investigating the issue by being empowered with clear guidelines and reference materials. I understand that puts additional work on the maintainers upfront, but the tradeoff is that the barrier to entry on recruiting contributors is lowered. I'm not opposed to deferring this work until a contributor indicates they're willing to work on it, but would the team be open to A/B testing this?

Perhaps we identify 6 issues, tag them as `help-wanted`, and work on providing some of the details outlined below for 3 of them. Then I will track all 6 of them and measure a few key metrics:

* Time to first claim
* Frequency of comments / communication in the issue / subsequent PR
* Time to PR opened
* Time to PR closed

File: help-wanted.md
Comment (kvncrw) [reply help-wanted.md]: > In the same way, I would just add a message when we apply the label with something like "We will have special attention and help you with this PR/issue, you can find some doc here and here, don't hesitate to ask for help".
And avoid writing another external document to read.

That's what the goal of the guidelines below are. That we prepare the issue accordingly to encourage new contributors and empower them with the details they need to start contributing. We should definitely timebox this effort, and be aware that not all these suggestions can always be met, but I would encourage the team to try and think about the bare minimum that is needed for new contributors to get rolling on an issue. (cf previous comment on help-wanted applies here as well)

File: help-wanted.md
Comment (kvncrw) [inline help-wanted.md]: take this out

**Review by rtribotte — APPROVED** [review]
LGTM 👌

**Review by kevinpollet — APPROVED** [review]
LGTM 👍

**Review by jspdown — APPROVED** [review]
LGTM

**Review by jbdoumenjou — APPROVED** [review]

**Pull request merged**
